src/appsettings/validators.py

```diff
@@ -67,3 +69,43 @@ def __call__(self, value):
             if not isinstance(element, self.value_type):
                 params = {"key": key, "value": element, "type": self.value_type.__name__}
                 raise ValidationError(self.message, params=params)
+
+
+class FileValidator(object):
+    """Validator which checks file existence and open mode permissions."""
+
+    not_exist_message = "File %(value)s does not exist."
+    no_perms_message = "You have no permission to open %(value)s with mode %(mode)s."
+    unexpected_error_message = "Unexpected error occurred while opening the file %(value)s: %(exception)s"
+
+    def __init__(
+        self, exists=True, mode="r", not_exist_message=None, no_perms_message=None, unexpected_error_message=None
+    ):
+        """
+        Try to open a file using a given ``mode``.
+
+        Args:
+            exists (bool): Require existence of the file
+            mode (str): ``mode`` argument for ``open`` builtin function.
+            not_exist_message (str): Error message (containing ``value``)
+            no_perms_message (str): Error message (containing ``value`` and ``mode``)
+            unexpected_error_message (str): Error message (containing ``value`` and ``exception``)
+        """
+        self.exists = exists
+        self.mode = mode
+        self.not_exist_message = not_exist_message or self.not_exist_message
+        self.no_perms_message = no_perms_message or self.no_perms_message
+        self.unexpected_error_message = unexpected_error_message or self.unexpected_error_message
+
+    def __call__(self, value):
+        try:
+            with open(value, mode=self.mode):
```

Comment:
One problem which I see is that it really opens and closes the file, so if there was a file with a content, this check removes the content no matter if you really use the file with write mode in your script. In this way the previous validation was better, safer (I mean the one with `os.access()`)

Reply:
Good point. That really doesn't look like something you want your settings should be doing. I would definitely avoid that.

Reply:
Should be all right now.